### DISCUSSION TITLE
Two Isolate glsp confict on Selection

### DISCUSSION THREAD

**Discussion by wss29** [eclipse-glsp/glsp#1370] [Q&A] [answered]
Hi,
    1. I have two glsp in my theia extensions,v1-glsp,v1-server,v1-theia. and v2-glsp,v2-server,v2-theia. 
    2. v1 used to deal with file format *.vdm, v2 used to deal  with file format *.vchip
    3. I open a vdm file and a vchip file
    4. the problem is that. When I click vchip title to select, the theiaSelectionService selection is vchip file, when I click vdm title to select, 
the theiaSelectionService selection is still vchip file.  are there any conflict when use two glsp in theia extensions?
![2024-07-16_10-09-43](https://github.com/user-attachments/assets/c78a4b5b-1eac-47d6-adb3-6ca73ce651b8)

**Comment by tortmayr** [comment]
HI @wss29,
looks like a bug to me. I can reproduce a similar behavior when using two .wf files in our examples so I don't think this is specifically related to a conflict when using two GLSP languages in Theia.

I have extracted a bug report from this discussion and we are looking into it.

**Comment by tortmayr** [comment, marked as answer]
A fix for this is provided with https://github.com/eclipse-glsp/glsp-theia-integration/pull/227
